### PR TITLE
Service alarms

### PR DESCRIPTION
**Pull request by PaloneAppEnjoyer**
please allow notifications in settings

### CODE REVIEW

File: app/src/main/java/com/palone/planahead/services/alarms/AlarmBroadcastReceiver.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            .setContentText("Nie zrobiłeś jeszcze tej rzeczy!").build()
+
+        if (alert != null) {
+            notificationManager.notify(
+                alert.alert_id!!,
+                if (alert.alert_type_name == AlertType.PERSISTENT_NOTIFICATION)
+                    persistentNotification
+                else singleNotification
+            )
+        }
```
Comment (Albitos) [inline app/src/main/java/com/palone/planahead/services/alarms/AlarmBroadcastReceiver.kt:62]:
```suggestion
        val notificationBuilder = NotificationCompat.Builder(context, channelId)
            .setContentTitle(task?.description)
            .setSmallIcon(R.drawable.ic_launcher_foreground)
            .setContentText("Nie zrobiłeś jeszcze tej rzeczy!");

        if (alert != null) {
            if (alert.alert_type_name == ) {
                notificationBuilder.setOngoing(true)
            }
            notificationManager.notify(
                alert.alert_id!!,
                notificationBuilder.build()
            )
        }
```

File: app/src/main/java/com/palone/planahead/services/alarms/AlarmsHandler.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.palone.planahead.data.database.task.Task
+import java.time.LocalDateTime
+import java.time.OffsetDateTime
+
+class AlarmsHandler(
+    private val context: Context,
+    private val alarmManager: AlarmManager
+) {
+    private var _tasks: Map<Task, List<Alert>> = emptyMap()
+    fun inflateTasks(tasks: Map<Task, List<Alert>>) {
```
Comment (Albitos) [inline app/src/main/java/com/palone/planahead/services/alarms/AlarmsHandler.kt:19]: Can't we just have tasks as `createAlarmEntries` param?

**Review by Albitos — APPROVED** [review]

File: app/src/main/java/com/palone/planahead/services/alarms/AlarmsHandler.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                        intent,
+                        PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+                    )
+                    alarmManager.setExactAndAllowWhileIdle(
+                        AlarmManager.RTC_WAKEUP,
+                        alert.alert_trigger_data?.toLong() ?: 0,
+                        pendingIntent
+                    )
+                    Log.i("Added :)", alert.toString())
+                }
```
Comment (Albitos) [inline app/src/main/java/com/palone/planahead/services/alarms/AlarmsHandler.kt:45]: Extract this block to a separate method.

File: app/src/main/java/com/palone/planahead/services/alarms/AlarmsHandler.kt (hunk 2 above)
Comment (PaloneAppEnjoyer) [reply app/src/main/java/com/palone/planahead/services/alarms/AlarmsHandler.kt:19]: Yes, this was a remnant of the testing phase

**Pull request merged**